components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/event/queue/EventListMapManager.java

```diff
@@ -152,7 +171,10 @@ public void trimAndSendToInputHandler() throws InterruptedException {
                     for (Source source : sources) {
                         if(queuedEvent.getSourceHandlerElementId().equals(source.getMapper().
                                 getHandler().getElementId())){
-                            source.getMapper().getHandler().sendEvent(queuedEvent.getEvent());
+                            source.getMapper().getHandler().sendEvent(queuedEvent.getEvent(),
+                                    null);
+                            ((HACoordinationSourceHandler)source.getMapper().getHandler()).
```

Comment:
All sources do not have HACoordinationSourceHandler, need to check if its a HACoordinationSourceHandler and then call this.

Reply:
Fixed